src/lib/base/secmem.h

```diff
@@ -21,7 +21,7 @@
 namespace Botan {
 
 template<typename T>
-class secure_allocator
```

Comment:
Maybe secure_allocator was designed to be base class for something? (or users of library may use it as base class for their own?)